src/checkers/inference/DefaultInferenceResult.java

```diff
@@ -0,0 +1,58 @@
+package checkers.inference;
+
+import checkers.inference.model.Constraint;
+
+import javax.lang.model.element.AnnotationMirror;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+public class DefaultInferenceResult implements InferenceResult {
```

Comment:
The interface `InferenceResult` contains javadoc about the returns of the top level methods. Can you add some javadoc here to clarify the differences between null vs empty set for varIdToAnnotation, and which constructors to call to properly create an inference result for no solution, empty solution, solution with no explanation, solution with explanation, etc?

Reply:
Done.